data_validation/query_builder/query_builder.py

```diff
@@ -201,7 +201,10 @@ def compile(self, ibis_table):
         comparison_field = ibis_table[self.field_name]
         alias = self.alias or self.field_name
         if self.cast:
-            comparison_field = comparison_field.force_cast(self.cast)
+            try:
+                comparison_field = comparison_field.force_cast(self.cast)
+            except Exception as e:
```

Comment:
We probably should catch the exception here since the 'force_cast()' method could throw a wide number of exceptions and we're assuming it's an invalid data type cast.

Instead let's add a try/except [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/third_party/ibis/ibis_addon/api.py#L46)

```
try:
        op = ops.Cast(self, to=target_type)
    except parsy.ParseError:
        raise SyntaxError(f"Invalid datatype provided in ComparisonField cast to {target_type}")
```

This will directly catch the ParseError that Ibis throws when a data type is invalid.